docs/admin/architectures/1k-users.md

```diff
@@ -0,0 +1,49 @@
+# Reference Architecture: up to 1,000 users
+
+The 1,000 users architecture is designed to cover a wide range of workflows.
+Examples of subjects that might utilize this architecture include medium-sized
+tech startups, educational units, or small to mid-sized enterprises.
+
+**Target load**: API: up to 180 RPS
+
+**High Availability**: non-essential for small deployments
+
+## Hardware recommendations
+
+### Coderd nodes
+
+| Users       | Node capacity       | Replicas | GCP             | AWS        | Azure             |
+| ----------- | ------------------- | -------- | --------------- | ---------- | ----------------- |
+| Up to 1,000 | 2 vCPU, 8 GB memory | 2        | `n1-standard-2` | `t3.large` | `Standard_D2s_v3` |
+
+**Footnotes**:
+
+- For small deployments (ca. 100 users, 10 concurrent workspace builds), it is
+  acceptable to deploy provisioners on `coderd` nodes.
+
+### Provisioner nodes
+
+| Users       | Node capacity        | Replicas                 | GCP              | AWS          | Azure             |
+| ----------- | -------------------- | ------------------------ | ---------------- | ------------ | ----------------- |
+| Up to 1,000 | 8 vCPU, 32 GB memory | 2 / 30 provisioners each | `t2d-standard-8` | `t3.2xlarge` | `Standard_D8s_v3` |
+
+**Footnotes**:
+
+- An external provisioner is deployed as Kubernetes pod.
+
+### Workspace nodes
+
+| Users       | Node capacity        | Replicas                | GCP              | AWS          | Azure             |
+| ----------- | -------------------- | ----------------------- | ---------------- | ------------ | ----------------- |
+| Up to 1,000 | 8 vCPU, 32 GB memory | 64 / 16 workspaces each | `t2d-standard-8` | `t3.2xlarge` | `Standard_D8s_v3` |
+
+**Footnotes**:
+
+- Assumed that a workspace user needs 2 GB memory to perform
```

Comment:
```suggestion
- Assumed that a workspace user needs at minimum 2 GB memory. We recommend against over-provisioning memory for developer workloads, as this my lead to OOMKiller invocations.
```

Reply:
Should we say _at most_ instead? If we exceed 2 GB/workspace, aren't we overprovisioning?

Reply:
I'm assuming that the 2GB here refers to the resources used by the workspace?
I can think of plenty of developer use-cases that need more than 2GB of RAM.

Reply:
I assumed that 2 GB is used here as an example. For people using `vim` or shared web hosting this would be great :)